### PR TITLE
Add Headphone Motion Manager

### PR DESCRIPTION
**Pull request by stephencelis**
This adds iOS 14's new `CMHeadphoneMotionManager` to `ComposableCoreMotion`. An experiment in incrementally improving these APIs even with beta APIs.

### CODE REVIEW

File: Sources/ComposableCoreMotion/HeadphoneMotionManager/HeadphoneMotionManagerInterface.swift [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import ComposableArchitecture
+import CoreMotion
+
+/// A wrapper around Core Motion's `CMHeadphoneMotionManager` that exposes its functionality through
+/// effects and actions, making it easy to use with the Composable Architecture, and easy to test.
```
Comment (mbrandonw) [inline Sources/ComposableCoreMotion/HeadphoneMotionManager/HeadphoneMotionManagerInterface.swift:5]: should we add more docs here about usage like we've done with other support libraries?

Comment (stephencelis) [reply Sources/ComposableCoreMotion/HeadphoneMotionManager/HeadphoneMotionManagerInterface.swift:5]: Yeah...I tried making a demo but it looks like this API doesn't even work yet: https://developer.apple.com/forums/thread/650051

Converted to a draft that we can revisit in later betas.

**Comment by andreyz** [comment]
Time to revisit and merge this piece of cool API?

**Comment by stephencelis** [comment]
Got around to testing this on a device and it works! Closing this draft, though, in favor of a new PR.

**Pull request closed without merging**